asmcli/commands/validate.sh

```diff
@@ -58,6 +52,12 @@ validate_dependencies() {
     fi
   fi
 
+  if can_register_cluster && is_gcp; then
```

Comment:
Cluster registration depends on GKE workload identity enablement. Thus move this section after enable_workload_identity.